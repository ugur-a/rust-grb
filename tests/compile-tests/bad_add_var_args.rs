use grb::prelude::*;
mod utils;

fn main() -> grb::Result<()> {
    create_model!(_g, m);
    add_var!(m, Binary, name: "x", name: "y")?;
    add_var!(m, Binary, name: "x", unknown: 30)?;
    add_var!(m, Binary, name: "x", bounds: ..=10)?;
    add_var!(m, Binary, name = "x")?;
    add_var!(m)?;
    Ok(())
}
