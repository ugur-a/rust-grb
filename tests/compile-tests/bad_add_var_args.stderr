error: duplicate argument
 --> tests/compile-tests/bad_add_var_args.rs:6:36
  |
6 |     add_var!(m, Binary, name: "x", name: "y")?;
  |                                    ^^^^

error: unknown argument 'unknown'
 --> tests/compile-tests/bad_add_var_args.rs:7:36
  |
7 |     add_var!(m, Binary, name: "x", unknown: 30)?;
  |                                    ^^^^^^^

error: Use '..' for bounds and range constraints
 --> tests/compile-tests/bad_add_var_args.rs:8:44
  |
8 |     add_var!(m, Binary, name: "x", bounds: ..=10)?;
  |                                            ^^^

error: expected `:`
 --> tests/compile-tests/bad_add_var_args.rs:9:30
  |
9 |     add_var!(m, Binary, name = "x")?;
  |                              ^

error: expected `,` (macro expects 2 positional args)
  --> tests/compile-tests/bad_add_var_args.rs:10:5
   |
10 |     add_var!(m)?;
   |     ^^^^^^^^^^^
   |
   = note: this error originates in the macro `add_var` (in Nightly builds, run with -Z macro-backtrace for more info)
